python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1042,10 +1046,38 @@ def execute_asset_backfill_iteration(
             )
             instance.update_backfill(updated_backfill)
 
+        new_materialized_partitions = (
+            updated_backfill_data.materialized_subset
+            - previous_asset_backfill_data.materialized_subset
+        )
+        new_failed_partitions = (
+            updated_backfill_data.failed_and_downstream_subset
+            - previous_asset_backfill_data.failed_and_downstream_subset
+        )
+        updated_backfill_in_progress = (
+            updated_backfill_data.requested_subset - updated_backfill_data.materialized_subset
+        )
+        previous_backfill_in_progress = (
+            previous_asset_backfill_data.requested_subset
+            - previous_asset_backfill_data.materialized_subset
+        )
+        new_requested_partitions = updated_backfill_in_progress - previous_backfill_in_progress
         logger.info(
             f"Asset backfill {updated_backfill.backfill_id} completed iteration with status {updated_backfill.status}."
         )
         logger.info(
+            "Backfill iteration summary:\n"
+            f"**Assets materialized since last iteration:**\n{_asset_graph_subset_to_str(new_materialized_partitions, asset_graph) if new_materialized_partitions.num_partitions_and_non_partitioned_assets > 0 else 'None'}\n"
+            f"**Assets failed since last iteration and their downstream assets:**\n{_asset_graph_subset_to_str(new_failed_partitions, asset_graph) if new_failed_partitions.num_partitions_and_non_partitioned_assets > 0 else 'None'}\n"
+            f"**Assets requested by this iteration:**\n{_asset_graph_subset_to_str(new_requested_partitions, asset_graph) if new_requested_partitions.num_partitions_and_non_partitioned_assets > 0 else 'None'}\n"
+        )
+        logger.info(
+            "Overall backfill status:\n"
+            f"**Materialized assets:**\n{_asset_graph_subset_to_str(updated_backfill_data.materialized_subset, asset_graph) if updated_backfill_data.materialized_subset.num_partitions_and_non_partitioned_assets > 0 else 'None'}\n"
+            f"**Failed assets and their downstream assets:**\n{_asset_graph_subset_to_str(updated_backfill_data.failed_and_downstream_subset, asset_graph) if updated_backfill_data.failed_and_downstream_subset.num_partitions_and_non_partitioned_assets > 0 else 'None'}\n"
+            f"**Assets requested or in progress:**\n{_asset_graph_subset_to_str(updated_backfill_in_progress, asset_graph) if updated_backfill_in_progress.num_partitions_and_non_partitioned_assets > 0 else 'None'}\n"
+        )
```

Comment:
* should we merge these log calls? not sure it matters
* should we avoid printing empty summaries / repeated overall statuses when we have ticks where nothing changes/progresses ?

Reply:
for empty summaries/statuses, i was thinking that knowing that nothing is happening/changed is still valuable. With a missing entry, you can infer that nothing has changed, but that requires more effort/memory of what could be logged than just having it told to you. Not sold that that's the right approach, but was my general line of reasoning

Reply:
> knowing that nothing is happening/changed is still valuable

I guess the line of thought is just communicating this more concisely and different than normal so that no-op ticks and progress ticks are more easily visually separated and the more frequent (at least in large long running backfills) no-op don't produce as much volume

Reply:
ah i see. gonna merge this as is and can modify in another pass. seeing the logs for long running backfills will be helpful